src/pudl/analysis/mcoe.py

```diff
@@ -220,13 +202,12 @@ def fuel_cost(pudl_out):
         ],
     ]
 
-    # We are inner merging here, which means that we don't get every generator
-    # in this output... we only get the ones that show up in hr_by_gen.
-    # See Issue #608
-    gen_w_ft = pudl.helpers.clean_merge_asof(
+    gen_w_ft = pudl.helpers.mixed_temporal_gran_merge(
         left=hr_by_gen,
         right=gens,
-        by=["plant_id_eia", "generator_id"],
+        shared_merge_cols=["plant_id_eia", "generator_id"],
+        temporal_merge_cols=["year"],
+        merge_type="left",
```

Comment:
The comment that was here for the call to `clean_merge_asof` says we're inner merging, but aren't we actually left joining? I made the merge a left join but I think inner and left join essentially do the same thing here?